### PR TITLE
refactor: more consolidation of box filters

### PR DESCRIPTION
**Pull request by ralfbrown**
This PR continues the process of consolidating all implementations of box filters in one source file by moving the box_average function from fast_guided_filters.h (used by tone equalizer with the old guided filter and focus peaking) as well as the previously-missed SSE codepath in soften.c.  It also optimizes some of the moved code.

The optimzation causes a different order of evaluation, which very slightly changes the output of test 0079 from the pending #7370 (304 total pixels changed, max dE = 1.0918, PAE equal to the minimum representable difference).

Benchmark results, using mire1-xtrans.raf to get longer runtimes than with mire1.cr2:
```
Soften (test 0051 sidecar)         ToneEq (GF) (test 0079 sidecar)
Thr   Old    New                        Thr   Old    New
1     3.799   2.378  (-36.4%)      1      1.436  1.048  (-27%)
4     0.987   0.640  (-35.1%)      4      0.398  0.297  (-25%)
8     0.540   0.389  (-28.0%)      8      0.220  0.164  (-25%)
16   0.329   0.282  (-14.3%)      16    0.123  0.096  (-22%)
32   0.260   0.256  (-1.5%)        32    0.084  0.079  (-6%)
```

### CODE REVIEW

**Comment by ralfbrown** [comment]
The bizzare M_LN10 not defined error in blend_gui.c again....  Is there a broken system header in the Win64 VM?

**Comment by johnny-bit** [comment]
By the look of #7378 I'd say it's generic problem in windows.

**Comment by TurboGit** [comment]
Thanks !

File: src/common/box_filters.c [hunk 1]
```diff
@@ -139,27 +141,31 @@ static void blur_vertical_1ch_sse(float *buf, const int height, const int width,
 #endif /* __SSE2__ */
 
 // invoked inside an OpenMP parallel for, so no need to parallelize
-static void blur_vertical_4wide(float *buf, const int height, const int width, const int radius, float *scanline)
+/*static*/ void blur_vertical_4wide(float *const restrict buf, const int height, const int width, const int radius,
+                                    float *const restrict scanline)
 {
 #ifdef __SSE2__
-  if (darktable.codepath.SSE2)
+  if (darktable.codepath.SSE2 & 0)
```
Comment (TurboGit) [inline src/common/box_filters.c]: @ralfbrown :
I see that now, was this expected ? Looks like debug code to disable SSE path ?

Comment (TurboGit) [reply src/common/box_filters.c]: I see, this was cleaned up in next commit, that's why I didn't see it.

Comment (ralfbrown) [reply src/common/box_filters.c]: Yep, debug code.

**Pull request merged**